datapackagist/src/scripts/components/ui/upload.js

```diff
@@ -20,7 +20,8 @@ module.exports = backbone.BaseView.extend({
           }
           catch(E) { }
 
-          if(!_.isObject(descriptor)) {
+          // If descriptor is broken or If descriptor have field not from schema - reject it
+          if(!_.isObject(descriptor) || _.difference(_.keys(descriptor), _.keys(this.parent.layout.form.schema.properties)).length) {
```

Comment:
@alexeyk it just checks if uploaded JSON has **all** fields from schema. It doesn't take in account ```.required``` property of a schema descriptor — https://rawgit.com/dataprotocols/schemas/master/data-package.json

![nzrlltrjnw7unkpfwymu-ar1iqctqngwmrd1n0allqu](https://cloud.githubusercontent.com/assets/58164/9730459/62d23430-561f-11e5-856e-da1c5dd5a3f3.png)

- [ ] Update the condition to check only required schema fields